rust_src/src/indent.rs

```diff
@@ -179,4 +180,124 @@ pub fn indent_to(column: EmacsInt, minimum: Option<EmacsInt>) -> EmacsInt {
     mincol
 }
 
+/// Scan through the current buffer, calculating screen position.
+/// Scan the current buffer forward from offset FROM, assuming it is
+/// at position FROMPOS--a cons of the form (HPOS . VPOS)-- to
+/// position TO or position TOPOS--another cons of the form (HPOS
+/// . VPOS)-- and return the ending buffer position and screen
+/// location.
+///
+/// If TOPOS is nil, the actual width and height of the window's text
+/// area are used.
+///
+/// There are three additional arguments:
+///
+/// WIDTH is the number of columns available to display text; this
+/// affects handling of continuation lines.  A value of nil
+/// corresponds to the actual number of available text columns.
+///
+/// OFFSETS is either nil or a cons cell (HSCROLL . TAB-OFFSET).
+/// HSCROLL is the number of columns not being displayed at the left
+/// margin; this is usually taken from a window's hscroll member.
+/// TAB-OFFSET is the number of columns of the first tab that aren't
+/// being displayed, perhaps because the line was continued within it.
+/// If OFFSETS is nil, HSCROLL and TAB-OFFSET are assumed to be zero.
+///
+/// WINDOW is the window to operate on.  It is used to choose the
+/// display table; if it is showing the current buffer, it is used
+/// also for deciding which overlay properties apply.  Note that
+/// `compute-motion' always operates on the current buffer.
+///
+/// The value is a list of five elements:
+///   (POS HPOS VPOS PREVHPOS CONTIN)
+/// POS is the buffer position where the scan stopped.
+/// VPOS is the vertical position where the scan stopped.
+/// HPOS is the horizontal position where the scan stopped.
+///
+/// PREVHPOS is the horizontal position one character back from POS.
+/// CONTIN is t if a line was continued after (or within) the previous character.
+///
+/// For example, to find the buffer position of column COL of line
+/// LINE of a certain window, pass the window's starting location as
+/// FROM and the window's upper-left coordinates as FROMPOS.  Pass the
+/// buffer's (point-max) as TO, to limit the scan to the end of the
+/// visible section of the buffer, and pass LINE and COL as TOPOS.
+#[lisp_fn]
+pub fn compute_motion(
+    from: LispNumber,
+    frompos: LispObject,
+    to: LispNumber,
+    topos: Option<LispObject>,
```

Comment:
LispObject is already essentially a `Option` type. No need to wrap it in one here. Check if `topos` is nil below and all should be fine.